### PR TITLE
fix(cvr, delete): ensure cvr gets deleted during reconcile

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This commit has following changes:
- remove the check where reconcile was avoided in case of previous delete failures
- reduce instances of log flooding
- improve log messages
- modify & indent code to make it more readable

### Related PR(s):
- https://github.com/openebs/maya/pull/1256
This commit tries to fix the symptom 2 talked in PR 1256

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go [hunk 1]
```diff
@@ -59,119 +60,178 @@ type CStorVolumeReplicaController struct {
...
+					ql.Operation = common.QOpSync
+
+					controller.recorder.Event(
+						newCVR,
+						corev1.EventTypeNormal,
+						string(common.SuccessSynced),
+						string(common.StatusSynced),
+					)
+
+				} else if IsDestroyEvent(newCVR) {
```
Comment (vishnuitta) [inline cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go]: is it good if we reverse this and the above 'if', i.e., how about below order:
```
if IsDestroyEvent(newCVR) {
} else if newCVR.ResourceVersion == oldCVR.ResourceVersion{
}else {
}
```

Comment (AmitKumarDas) [reply cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go]: what is the motivation for above? @vishnuitta

Comment (AmitKumarDas) [reply cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go]: Understood. Done

File: cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go [hunk 2]
```diff
@@ -59,119 +60,201 @@ type CStorVolumeReplicaController struct {
...
+			// **if** conditions. Only the reject related conditions
+			// should be here. Other conditions should be part
+			// of actual business logic that handles change to
+			// a resource.
+			UpdateFunc: func(old, new interface{}) {
+				newCVR := new.(*apis.CStorVolumeReplica)
+				oldCVR := old.(*apis.CStorVolumeReplica)
+
+				glog.V(4).Infof(
+					"received informer update event for cvr {%s}",
```
Comment (mynktl) [inline cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go]: can we log this after `IsRightCStorVolumeReplica` check?

Comment (AmitKumarDas) [reply cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go]: ok

Comment (AmitKumarDas) [reply cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go]: done

Comment (vishnuitta) [reply cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go]: :+1:

**Pull request merged**